opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/FragmentBaseActivity.java

```diff
@@ -0,0 +1,279 @@
+            fo.write(bytes.toByteArray());
+        } catch (IOException e) {
+            Timber.e(e);
+        }
+    }
+
+
+    private void sendEmailWithAttachment(String subject, String pathToMyAttachedFile) {
+        Intent emailIntent = new Intent(Intent.ACTION_SEND);
+        emailIntent.setType("text/plain");
+        emailIntent.putExtra(Intent.EXTRA_SUBJECT, subject);
+        // emailIntent.putExtra(Intent.EXTRA_TEXT, " ");
+        File file = new File(pathToMyAttachedFile);
+        if (!file.exists() || !file.canRead()) {
+            Timber.e("File does not exist");
+            return;
+        }
+        Uri uri = FileProvider.getUriForFile(this, getApplicationContext().getPackageName() + ".fileprovider", file);
+        emailIntent.putExtra(Intent.EXTRA_STREAM, uri);
+        startActivity(Intent.createChooser(emailIntent, "Pick an Email provider"));
+    }
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_fragment_base);
+        Toolbar toolbar = findViewById(R.id.toolbar_top);
+        setSupportActionBar(toolbar);
+
+        titleTextView = findViewById(R.id.toolbar_title);
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setDisplayShowHomeEnabled(true);
+
+            final Drawable upArrow = getResources().getDrawable(R.drawable.ic_arrow_back_white_24dp);
+            upArrow.setColorFilter(getResources().getColor(R.color.white), PorterDuff.Mode.SRC_ATOP);
+            actionBar.setHomeAsUpIndicator(upArrow);
+            actionBar.setElevation(0);
+        }
+
+        toolbar.setNavigationOnClickListener(v -> onBackPressed());
+
+        Bundle bundle = getIntent().getExtras();
+        if (bundle != null) {
+            String title = bundle.getString(TITLE);
+            if (StringUtils.isNotBlank(title)) {
+                titleTextView.setText(title);
+            }
+            INDICATOR_CODE = bundle.getString(INDICATOR_CODE);
+
+            String report_date = bundle.getString(GizConstants.ReportParametersHelper.REPORT_DATE);
+            String communityName = bundle.getString(GizConstants.ReportParametersHelper.COMMUNITY);
+            String fragmentName = bundle.getString(DISPLAY_FRAGMENT);
+            String name = getRequestedFragmentName(fragmentName);
+
+            //Format Subject Date
+            if (report_date != null) {
+                DateFormat df = new SimpleDateFormat("dd MMM yyyy");
+                Date date = null;
+                try {
+                    date = df.parse(report_date);
+                } catch (ParseException e) {
+                    Timber.e(e, "Date could not be formatted");
+                }
+
+                SimpleDateFormat sdfTime = new SimpleDateFormat("hh:mm:ss", Locale.US);
```

Comment:
Could these DateTIme formats be organized on the constants?

Reply:
Done